src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -1,21 +1,20 @@
 package cromwell.engine.db
 
-import cromwell.engine.WorkflowOutputs
-import wdl4s._
 import cromwell.engine.ExecutionStatus.ExecutionStatus
-import cromwell.engine._
+import cromwell.engine.{WorkflowOutputs, _}
 import cromwell.engine.backend.{Backend, JobKey}
 import cromwell.engine.db.slick._
 import cromwell.engine.workflow.{BackendCallKey, ExecutionStoreKey, OutputKey}
 import cromwell.webservice.{CallCachingParameters, WorkflowQueryParameters, WorkflowQueryResponse}
+import wdl4s._
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 object DataAccess {
   val globalDataAccess: DataAccess = new slick.SlickDataAccess()
 }
 
-trait DataAccess {
+trait DataAccess extends AutoCloseable {
```

Comment:
I haven't really seen anyone use this in scala, presumably just because it's a new-ish construct. I'm not saying this to poo poo it (it seems like a good idea), but more just to ensure we understand how it's actually working here. Is this something you've seen other slick stuff doing?